cnd/tests/ethereum_create_contract_matching.rs

```diff
@@ -12,14 +14,14 @@ use web3::{
     Web3,
 };
 
-/// A very basic e2e test that verifies that we glued all our code together
-/// correctly for ethereum transaction pattern matching.
-///
-/// We get the default account from the node and send some money to it
-/// from the parity dev account. Afterwards we verify that the tx hash of
-/// the sent tx equals the one that we found through btsieve.
+const VALUE: u32 = 2_000_000_000;
+
+lazy_static::lazy_static! {
+    static ref REDEEM_LOG_MSG: H256 = blockchain_contracts::ethereum::rfc003::REDEEMED_LOG_MSG.parse().expect("to be valid hex");
+}
+
 #[tokio::test]
-async fn ethereum_transaction_pattern_e2e_test() {
```

Comment:
I don't agree with repurposing this test.

The original idea was:

- have fast (i.e. non-IO) integration tests of the whole btsieve module by using a connector mock and real data (that's the tests you deleted for now). they verify things like missing blocks, etc but are unconcerned about the actual matching (they don't need to as it is orthogonal)
- have unit tests for the actual matching logic (with the new design, that might be obsolete but I guess it wouldn't hurt if we have some basic tests for edge-case logic). You can pass function pointers instead of closures and unit test the functions you are pointing to.
- have a **single**, slow e2e test that verifies that we glued things together correctly (this test). Note that this test should be as simple as possible and not use complicated contracts like HTLCs or something.

All these three topics are orthogonal and together form a good test coverage without introducing unnecessary dependencies and levels of detail.

The added value of these tests is very little because we have the e2e tests anyway and they cover the same thing (they also use actual HTLCs).
Yet, they are super heavy and slow.

What we need to test at this level of the code are edgecases that are _not_ covered by the e2e tests:

- missing blocks
- go back in the past
- etc

but those are missing with your testing strategy.
I'd recommend to revert this and just bring the other tests back.

Reply:
Ok, I think I've got you.  This is what I've understood to do

- Add unit tests that test the private matching functions i.e., use different matching closures
- Add integration tests that test the API, use static data for these (you didn't say this but as a precursor to the next point I think this is useful)
- Add integration tests that test 'go back in the past' and 'missing blocks'
- Remove the parity container integration test (you didn't say this but the description of point 3 for the old design is no longer valid/useful with the new design)

@thomaseizinger I've pushed a couple of commits that add skeleton functions.  I'm going to go ahead and work on these, if you feel like taking a look at the direction I'm taking please do so.  I do not think that the added integration test (using parity) adds much although it is still included because its done and working already, can probably be removed before this PR merges.  Thanks

Reply:
In case you saw the message above mid-edit, I had this message sitting open for an hour and during which I wrote the skeleton test functions.  Seems I contradicted myself.